src/routes/friend.js

```diff
@@ -1,88 +1,164 @@
-});
+/**
+ * @author: Jasmin Guay
+ * Endpoint to update a student friend list
+ * @param {string} email : email of the student to be updated
+ * @param {string} friends : list of updated friend list.
+ * @return {[string]} the update friend list.
+ */
+router.route('/updateFriendList').post( async (req,res) => {
+    const email = req.body.email.toString();
+    const updatedFriendList = req.body.friends.toString();
+
+    StudentRepository.updateFriendList(email,updatedFriendList)
+        .then((student) => { res.json(student.friends).status(200) })
+        .catch((err) => { res.json(`Error happened in updateFriendList - ${err}`).status(404) })
+})
 
-router.route('/answer').post(async (req, res) => {
 
-  const username = req.body.username.toString();
-  const friendUsername = req.body.friendUsername
-  const answer = req.body.answer;
+/**
+ * @author: Jasmin Guay
+ * Endpoint to answer a friend request.
+ * @param {string} requestId  id of the request.
+ * @param {"accepted" | "declined"} answer : answer to the friend request.
+ * @return {string || null} studentUsername : username of the student added to the friend list if the request was accepted, null otherwise.
+ */
+router.route('/answerFriendRequest').post(async (req, res) => {
+  const requestId = req.body.requestId.toString();
+  const user = req.body.user;
+  const request = await FriendRequest.findOne({_id: requestId});
 
-  Student.updateOne({ username: username },
-    { $pull: { friendRequestsReceived: { $in: [friendUsername] } } })
+  if (!validateFriendRequest(request, user) || request == null) {
+    res.json(`No request found with id [${requestId}`).status(404);
+  } else {
+    if (req.body.answer === "accepted") {
+      await addToFriendLists(request.senderEmail, request.receiverEmail);
+      res.json(request.receiverEmail).status(200);
+    } else {
+      res.status(200)
+    }
 
+    await FriendRequestRepository.delete(requestId);
+  }
 });
 
-// //upload a file to the database not yet complete
-// router.route('/file').post(async(req, res) => {
-
-//     let notes;
-//     let r = (Math.random() + 1).toString(36).substring(7);
-//     const roomID  = req.body.sID; 
-//     const file =  req.body.file;
-//     const username = req.body.username;
-//      console.log(roomID)
-//     const room = await StudyRoom.findOne({
-//         sID:roomID,
-//       });
-//     let note = {
-//          nID:r,
-//          username:username,
-//          file:file
-//       }
-//       console.log(room);
-//       notes = room.courseNotes
-//       notes.push(note)
-//       StudyRoom.updateOne(
-//         { sID: roomID },
-//         { courseNotes: notes },
-//         (err, docs) => {
-//           if (err) {
-//             console.log(err);
-//           } else {
-//             console.log('Updated Docs : ', docs);
-//           }
-//         },
-//       );
-// });
+function validateFriendRequest(request, user) {
+  return user.email === request.senderEmail.toString();
```

Comment:
This should be `receiverEmail`. The receiver should be the one to accept the friend request, not the sender.